src/overlays/actors/ovl_En_Bubble/z_en_bubble.c

```diff
@@ -23,45 +26,393 @@ const ActorInit En_Bubble_InitVars = {
+u32 EnBubble_DetectPop(EnBubble* this, GlobalContext* globalCtx) {
+    if (DECR(this->unk_208) != 0 || this->actionFunc == EnBubble_Pop) {
+        return 0;
+    }
+    if (this->colliderSphere.base.maskB & 0x1) {
+        this->colliderSphere.base.maskB &= ~0x1;
+        EnBubble_DamagePlayer(this, globalCtx);
+        this->unk_208 = 8;
+        return 1;
+    }
+    return func_809CC648(this);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Bubble/func_809CBD10.s")
+void func_809CC774(EnBubble* this) {
+    ColliderJntSphItemDim* dim;
+    Vec3f src;
+    Vec3f dest;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Bubble/func_809CBD78.s")
+    dim = &this->colliderSphere.list[0].dim;
+    src.x = dim->modelSphere.center.x;
+    src.y = dim->modelSphere.center.y;
+    src.z = dim->modelSphere.center.z;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Bubble/func_809CBFD4.s")
+    Matrix_MultVec3f(&src, &dest);
+    dim->worldSphere.center.x = dest.x;
+    dim->worldSphere.center.y = dest.y;
+    dim->worldSphere.center.z = dest.z;
+    dim->worldSphere.radius = dim->modelSphere.radius * (1.0f + this->expansionWidth);
+    this->colliderSphere.list[1].dim = *dim;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Bubble/func_809CC020.s")
+void EnBubble_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnBubble* this = THIS;
+    u32 pad;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Bubble/func_809CC074.s")
+    ActorShape_Init(&this->actor.shape, 16.0f, ActorShadow_DrawFunc_Circle, 0.2f);
+    Collider_InitJntSph(globalCtx, &this->colliderSphere);
+    Collider_SetJntSph(globalCtx, &this->colliderSphere, &this->actor, &sJntSphInit, this->colliderSphereItems);
+    func_80061EFC(&this->actor.colChkInfo, DamageTable_Get(9), &sColChkInfoInit2);
+    this->actor.naviEnemyId = 0x16;
+    this->bounceDirection.x = Rand_ZeroOne();
+    this->bounceDirection.y = Rand_ZeroOne();
+    this->bounceDirection.z = Rand_ZeroOne();
+    EnBubble_Vec3fNormalize(&this->bounceDirection);
+    this->velocityFromBounce.x = this->bounceDirection.x * 3.0f;
+    this->velocityFromBounce.y = this->bounceDirection.y * 3.0f;
+    this->velocityFromBounce.z = this->bounceDirection.z * 3.0f;
+    EnBubble_SetDimensions(this, 0.0f);
+    this->actionFunc = EnBubble_Wait;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Bubble/func_809CC0FC.s")
+void EnBubble_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnBubble* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Bubble/func_809CC16C.s")
+    Collider_DestroyJntSph(globalCtx, &this->colliderSphere);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Bubble/func_809CC648.s")
+void EnBubble_Wait(EnBubble* this, GlobalContext* globalCtx) {
+    if (EnBubble_DetectPop(this, globalCtx) != 0) {
+        this->explosionCountdown = func_809CBCBC(this);
```

Comment:
seems to be a boolean, you can remove the !=0 and change the function to return true and false